pioneer.py

```diff
@@ -104,42 +106,88 @@ def main():
                 stop_game = True
             
             #### Hero Movement
+            
             if event.type == pygame.KEYDOWN:
                 if event.key == pygame.K_LEFT:
                     hero.speed_x = -5
+                    rect_x = -5
+                    # print hero_rect.colliderect(mountain)
```

Comment:
I found this on Stack Overflow that may help with your collisions.
https://stackoverflow.com/questions/44721130/pygame-collision-detection-with-walls